### PR TITLE
feat: add application image for snapshots

### PR DESCRIPTION
**Pull request by oap75**

### CODE REVIEW

File: charts/rollups-node/values.yaml.tpl [hunk 1]
```diff
@@ -64,13 +64,22 @@ ingress:
 extraDeploy: []
 
 validator:
-  # -- Set the dispatcher docker image
+  # -- Set the validator docker image
```
Comment (endersonmaia) [inline charts/rollups-node/values.yaml.tpl:67]: This fix should be on its own commit.

File: charts/rollups-node/ci/test-values.yaml.tpl [hunk 2]
```diff
@@ -22,7 +26,7 @@ validator:
     CARTESI_FEATURE_DISABLE_MACHINE_HASH_CHECK: "true"
     CARTESI_SNAPSHOT_DIR: "/usr/share/cartesi/snapshot"
   initContainers:
-    - image: "docker.io/cartesi/dapp:echo-python-0.16.0-server"
+    - image: "{{ .Values.validator.application.image.repository }}:{{ .Values.validator.application.image.tag }}"
```
Comment (endersonmaia) [inline charts/rollups-node/ci/test-values.yaml.tpl:29]: we should reuse the helper method at https://github.com/cartesi/helm-charts/blob/main/charts/rollups-node/templates/_helpers.tpl#L107-L121

or create a wrapper like https://github.com/cartesi/helm-charts/blob/main/charts/rollups-node/templates/_helpers.tpl#L137-L142

so we could use `{{ include "application.image" . }}` at the end

**Review by endersonmaia — CHANGES_REQUESTED** [review]

File: charts/rollups-node/ci/test-values.yaml.tpl (hunk 2 above)
Comment (oap75) [reply charts/rollups-node/ci/test-values.yaml.tpl:29]: the include function should typically be used within template files rather than directly within the YAML structure of `values.yaml`.

> so we could use `{{ include "application.image" . }}` at the end

Comment (oap75) [reply charts/rollups-node/ci/test-values.yaml.tpl:29]: I think, we can use this approach to make it simpler : 

```
validator:
  application:
    image: cartesi/dapp:echo-python-0.16.0-server

```

And then fetch the value: 

```
validator:
  initContainers:
    - image: "{{ .Values.validator.application.image }}"
```

Comment (oap75) [reply charts/rollups-node/ci/test-values.yaml.tpl:29]: OR we can remove the `- image` section from `initcontainer` of values.yaml to the deployment template. 
I'll send a fixup for it.

Comment (oap75) [reply charts/rollups-node/ci/test-values.yaml.tpl]: https://github.com/cartesi/helm-charts/pull/83/commits/3179ad62a6b1565498d578b9ddd22e494e36e994

https://github.com/cartesi/helm-charts/pull/83/commits/d5150da3978203c6e06f31641e7ff05083e5cf3f

File: charts/rollups-node/templates/deployment.yaml [hunk 3]
```diff
@@ -33,6 +33,7 @@ spec:
       initContainers:
         {{- if .Values.validator.initContainers }}
         {{- include "tplvalues.render" ( dict "value" .Values.validator.initContainers "context" $ ) | nindent 8 }}
+          image: {{ include "application.image" . }}
```
Comment (endersonmaia) [inline charts/rollups-node/templates/deployment.yaml:36]: How is this gonna work?

`initContainers` is a list, we don't know how many `initContainer` the user is gonna define.

For ex. we may have an `initContainer` for the snapshot and to create the database.

**Review by endersonmaia — CHANGES_REQUESTED** [review]

File: charts/rollups-node/templates/deployment.yaml (hunk 3 above)
Comment (endersonmaia) [reply charts/rollups-node/templates/deployment.yaml:36]: We shouldn't force a naming, the user could provide a snapshot any other way without even using an `initContainer`, for ex.

The approach in this https://github.com/cartesi/helm-charts/pull/83#discussion_r1482480224 is the easy path for now.

**Review by endersonmaia — CHANGES_REQUESTED** [review]

File: charts/rollups-node/README.md [hunk 4]
```diff
@@ -67,12 +67,10 @@ The command removes all the Kubernetes components associated with the chart and
 
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
-| dapp.image | string | `nil` | the complete image name, e.g. "cartesi/dapp:echo-python-0.8.0-server" (REQUIRED) |
-| dapp.network | string | `nil` | the name of the network the dapp is deployed on (REQUIRED) Available options are:   - mainnet   - optimism   - optimism-goerli   - arbitrum   - arbitrum-goerli   - localhost   - sepolia |
 | extraDeploy | list | `[]` | Array of extra objects to deploy with the release |
 | fullnameOverride | string | `""` | String to fully override name |
 | global.image.registry | string | `"docker.io"` | Global Docker image registry |
-| global.image.tag | string | `"1.3.0"` | Global Docker Image tag |
+| global.image.tag | string | `"1.3.0-rc.1"` | Global Docker Image tag |
```
[the remark this answers is not in the document]
Comment (oap75) [reply charts/rollups-node/README.md:73]: there is no 1.3.0 docker version anymore:

```bash
docker pull docker.io/cartesi/rollups-node:1.3.0
```

image

File: charts/rollups-node/README.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 # Package for Cartesi Rollups Nodes
 
-![Version: 1.3.0-0](https://img.shields.io/badge/Version-1.3.0--0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square)
+![Version: 1.3.0-rc.1](https://img.shields.io/badge/Version-1.3.0--rc.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square)
```
[the remark this answers is not in the document]
Comment (oap75) [reply charts/rollups-node/README.md:5]: This will show the current PR action status. So, In My Idea, Every PR should have it's own README based on the charts and versions. So, when we merge this PR, we should regenerate the README based on final PR.

[the remark this answers is not in the document]
Comment (oap75) [reply charts/rollups-node/README.md:5]: image

This picture is related to README of this PR.

File: charts/rollups-node/values.yaml.tpl [hunk 6]
```diff
@@ -8,7 +8,7 @@ global:
     # -- Global Docker image registry
     registry: docker.io
     # -- Global Docker Image tag
-    tag: 1.3.0
+    tag: 1.3.0-rc.1
```
[the remark this answers is not in the document]
Comment (endersonmaia) [reply charts/rollups-node/values.yaml.tpl:11]: There's no 1.3.0 yet. That's why we keep using a `1.3.0-rc.1` ina `DELETE:` commit so the CI passes, until we have a final 1.3.0 release an then we remove this commit and move on.

**Review by endersonmaia — APPROVED** [review]
